app/services/resource_remover.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# service class for removing a resource or multiple resources
+class ResourceRemover
+  def self.remove_all_resources
+    new.remove_all_resources
+  end
+
+  # @return [Integer] the number of records removed
+  def remove_all_resources
+    # we could run a count query, but if there are a ton of records, actual deleted count could
+    # exceed initial count, because find_each runs in batches non-transactionally.
+    removal_count = 0
+    Spotlight::Resource.find_each do |resource|
+      ResourceRemover.remove_resource(resource: resource)
+      removal_count += 1
+    end
+    removal_count
+  end
+
+  def self.remove_resource(resource:)
+    new.remove_resource(resource: resource)
+  end
+
+  # @param [Spotlight::Resource] resource to be un-indexed and destroyed
+  def remove_resource(resource:)
```

Comment:
i didn't add tests for this method to the new service class' specific tests, since the existing `DeleteResourcesJob` tests cover it, but can revisit that if others think that'd be a good idea.

Reply:
the new `remove_all_resources` test implicitly exercises this too